network/src/commonMain/kotlin/com/wire/kalium/network/api/base/authenticated/notification/EventContentDTO.kt

```diff
@@ -294,7 +299,60 @@ sealed class EventContentDTO {
         ) : FeatureConfig()
     }
 
+    @Serializable
+    sealed class UserProperty : EventContentDTO() {
+        @Serializable
+        @SerialName("user.properties-set")
+        data class PropertiesSetDTO(
+            @SerialName("key") val key: String,
+            @SerialName("value") val value: FieldKeyValue,
+        ) : UserProperty()
+
+        @Serializable
+        @SerialName("user.properties-delete")
+        data class PropertiesDeleteDTO(
+            @SerialName("key") val key: String,
+        ) : UserProperty()
+
+        enum class PropertyKey(val key: String) {
+            WIRE_RECEIPT_MODE("WIRE_RECEIPT_MODE")
+            // TODO map other event like -ie. 'labels'-
+        }
+    }
+
+    @Serializable(with = FieldKeyValueDeserializer::class)
+    sealed interface FieldKeyValue
+
+    @Serializable
+    @JvmInline
+    value class FieldKeyNumberValue(val value: Int) : FieldKeyValue
+
+    @Serializable
+    @JvmInline
+    value class FieldUnknownValue(val value: String) : FieldKeyValue
+
     @Serializable
     @SerialName("unknown")
     object Unknown : EventContentDTO()
 }
+
+@OptIn(ExperimentalSerializationApi::class, InternalSerializationApi::class)
+@Serializer(EventContentDTO.FieldKeyValue::class)
+class FieldKeyValueDeserializer : KSerializer<EventContentDTO.FieldKeyValue> {
```

Comment:
```suggestion
object FieldKeyValueDeserializer : KSerializer<EventContentDTO.FieldKeyValue> {
```

just to avoid creating endless instances of this  serializer